lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
+      }
 
-    if (!fieldSpec.multivalued.value && (fieldSpec.multivalued.hasOwnProperty('minCount'))) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "minCount" is not valid when multivalued field is disabled.`, verboseErrors, errors);
-    }
+      if (!fieldSpec.multivalued.value && (fieldSpec.multivalued.hasOwnProperty('maxCount'))) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "maxCount" is not valid when multivalued field is disabled.`, verboseErrors, errors);
+      }
 
-    if (!fieldSpec.multivalued.value && (fieldSpec.multivalued.hasOwnProperty('maxCount'))) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "maxCount" is not valid when multivalued field is disabled.`, verboseErrors, errors);
+      if (fieldSpec.multivalued.hasOwnProperty('minCount') && fieldSpec.multivalued.hasOwnProperty('maxCount') && fieldSpec.multivalued.minCount > fieldSpec.multivalued.maxCount) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "minCount" can not be greater than "maxCount".`, verboseErrors, errors);
+      }
     }
 
-    if (fieldSpec.multivalued.hasOwnProperty('minCount') && fieldSpec.multivalued.hasOwnProperty('maxCount') && fieldSpec.multivalued.minCount > fieldSpec.multivalued.maxCount) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "minCount" can not be greater than "maxCount".`, verboseErrors, errors);
+    if (errors.length > 0) {
+      return {isValid: false, errors: errors};
     }
+    return {isValid: true};
   }
 
-  if (errors.length > 0) {
-    return {isValid: false, errors: errors};
-  }
-  return {isValid: true};
-};
+  /**
+   * @param {string} indexName
+   * @param {string} collectionName
+   * @param {*} validatorFilter
+   * @param {boolean} dryRun
+   * @returns {Promise}
+   */
+  curateValidatorFilter(indexName, collectionName, validatorFilter, dryRun) {
+    var
+      promise,
+      query = {
+        bool: {
+          must: validatorFilter
+        }
+      };
 
-/**
- * @param {string} indexName
- * @param {string} collectionName
- * @param {*} validatorFilter
- * @param {boolean} dryRun
- * @returns {Promise}
- */
-Validation.prototype.curateValidatorFilter = function validationCurateValidatorFilter (indexName, collectionName, validatorFilter, dryRun) {
-  var
-    promise,
-    query = {
-      bool: {
-        must: validatorFilter
-      }
-    };
+    promise = this.dsl.validate(query);
 
-  promise = this.dsl.validate(query);
+    if (!dryRun) {
+      promise.then(() => {
+        return this.dsl.register(indexName, collectionName, query);
+      });
+    }
 
-  if (!dryRun) {
-    promise.then(() => {
-      return this.dsl.register(indexName, collectionName, query);
-    });
+    return promise;
   }
 
-  return promise;
-};
+  /**
+   * @param {ValidationType} validationType
+   * @throws {PluginImplementationError}
+   */
+  addType(validationType) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AFunctionComplexity)